### DISCUSSION TITLE
Question: Does parallel plotting makes sense in this plotter?

### DISCUSSION THREAD

**Discussion by delucca** [madMAx43v3r/chia-plotter#92] [General]
Hi !

First and foremost, I've run a few tests yesterday with your plotter. The speed is impressive! My rig takes ~6h to plot using Chia default plotter, and with your library, it took only 1h10min (6x faster).

Anyway, during my tests yesterday I tried two scenarios:
- Serial plotting
- Parallel plotting

While serial plotting, it took 1h10min, but while parallel plotting (with proper stagger minutes) it took 3h30min (running 3/4 plots at the same time)

So, my question is, since we assume this plotter optimizes the resource usage. Does it make sense to try to parallel plot? In my simple test scenario, it seems that while parallel plotting the plot time increased at the same rate. So, IMHO, it seems that we optimize the plotting so much that parallel plotting doesn't make sense.

I would like to know your opinion regarding this.

**Comment by mist0706** [comment]
Short answer: No

Here is the CPU usage graph for one of my 5950x rigs:
![image](https://user-images.githubusercontent.com/10143260/121354614-54e6f480-c92f-11eb-86be-43ea5932f7ee.png)

What we can deduct from it is that phase1 is already very optimized cpu wise, it uses up pretty much the entire CPU. 
Phase 2 also uses most of the cpu however at intervals
Phase 3/4 are the least cpu intensive tasks.

The only way i can see to squeeze out a tiny bit more performance is to start a new plot as soon as phase 3 is done. This will only net you about 60s of parallelism in my case. You might be able to launch a plot when phase2 is done but you'll have issues with CPU context switching which might render it meaningless.

**Reply by rHilkner** [reply]
Hey, I also have a 5950x. What are your NVME drives? Do they ever go to 100% util? How many plots are you managing to make a day?
It would be great if you could share your configs as well, I'm curious to test this project!

**Reply by delucca** [reply]
I have a 5900x with a 970EVO, a k32 plot here takes ~1h, so I can make ~24 plots a day (with chia-plotter)

**Comment by delucca** [comment]
Great answer! Thanks @mist0706 

That was exactly what I was expecting. I was not able to gather formal results regarding CPU usage yet, but my feeling was exactly that.

IMHO the only way we could parallelize to improve performance would be by implementing MPI or another parallel API to distribute CPU work across multiple nodes.

Quick question, have you installed Prometheus to gather metrics regarding your rigs? That is a pretty good idea :)

**Comment by madMAx43v3r** [comment]
Yeah overlapping phase 1 and 3 is your best bet, how much you'll gain depends on your `tmpdir` disk speed.

**Reply by andymarden** [reply]
I've tried it and I got, I reckon about 1-2 mins improvement on average on 62 mins plotting time with one "queue" (only running an NUC10). Not worth it imho - for breaking away from that lovely predictability where the plots pop out like clockwork and you can lose very little time even if you reboot after a plot is finished.

I am glad in way - it's a great plotter and the simplicity & predictability is as much of a benefit as the performance.

**Comment by delucca** [comment]
In any case, given the amazing performance this plotter have, I really think that the effort to find the exact stagger time to overlap phase 1 and 3 don't pay off.

I've done that with chia default plotter, and it is pretty hard to find the exact time.

**Comment by madMAx43v3r** [comment]
It'll be an option in the plotter itself (at some point), to run 2 or 3 in parallel staggered.

**Reply by Mar-85** [reply]
Great news, are you still looking to have this work with the 110G ramdisk or for this feature we'll need to double the RAM or add an second temp SSD/NVME?

**Comment by delucca** [comment]
Well, if the plotter itself can do that, it would be pretty amazing!

**Comment by digitalscream** [comment]
Short answer: It depends.

Long answer: On my dual E5-2450 machine, the fastest I can get with my available hardware is sub-2hr plots, using all 16 cores, SAS RAID0 for tmp1 and NVMe for tmp2. That works out at 12-13 plots per day.

However, restricting it to 4 cores per plot process, with 4 x 4 disk SAS RAID0 arrays arranged in tmp1/tmp2 pairs, and using taskset to force each process to only run on the physical and logical cores of half of each physical CPU, I can run 4 of these plotters in parallel at ~2h50m per plot, thus netting 32+ plots per day. Or, to put it another way, plot output equivalent to 45 minute plots on a single process.

It's about balancing your hardware more than getting the absolute fastest single-plot time.

**Reply by ndaskalovic** [reply]
What is the speed of your raid array? I have a 5900x and a 4tb NVMe SSD and im trying to figure out how to run two instances of the plotter in parallel. Starting at the same time I get only 10% performance gain compared to two plots in serial. 45 mins for on in serial and 83 for two in parallel. I also have several SATA hard drives that I could put into raid 0.

**Reply by mostafa-aw** [reply]
> Short answer: It depends.
> 
> Long answer: On my dual E5-2450 machine, the fastest I can get with my available hardware is sub-2hr plots, using all 16 cores, SAS RAID0 for tmp1 and NVMe for tmp2. That works out at 12-13 plots per day.
> 
> However, restricting it to 4 cores per plot process, with 4 x 4 disk SAS RAID0 arrays arranged in tmp1/tmp2 pairs, and using taskset to force each process to only run on the physical and logical cores of half of each physical CPU, I can run 4 of these plotters in parallel at ~2h50m per plot, thus netting 32+ plots per day. Or, to put it another way, plot output equivalent to 45 minute plots on a single process.
> 
> It's about balancing your hardware more than getting the absolute fastest single-plot time.



how do u use taskset to assume CPU cores?

**Reply by SebMoore** [reply]
> > Short answer: It depends.
> > Long answer: On my dual E5-2450 machine, the fastest I can get with my available hardware is sub-2hr plots, using all 16 cores, SAS RAID0 for tmp1 and NVMe for tmp2. That works out at 12-13 plots per day.
> > However, restricting it to 4 cores per plot process, with 4 x 4 disk SAS RAID0 arrays arranged in tmp1/tmp2 pairs, and using taskset to force each process to only run on the physical and logical cores of half of each physical CPU, I can run 4 of these plotters in parallel at ~2h50m per plot, thus netting 32+ plots per day. Or, to put it another way, plot output equivalent to 45 minute plots on a single process.
> > It's about balancing your hardware more than getting the absolute fastest single-plot time.
> 
> how do u use taskset to assume CPU cores?

For example:
`taskset -c 1-8 ./chia-plotter [insert the options you want here as usual]`
would run chia-plotter with the options you specify on cores 1-8 of your system.

**Reply by mostafa-aw** [reply]
In the madmax script? 
I'm new to these powershell commands. Can u share ur format??

**Reply by digitalscream** [reply]
@mostafa-aw - this isn't Windows, it's Linux.

This is what I'm using, on a 2 x 8 core Xeon machine with SAS RAID0 as tmp2 and my SAS RAID5 destination array as tmp1:
 
taskset -ac 0-15 ~/bin/chia-plotter/build/chia_plot <keys> -t /mnt/plots_8/tmp_1/ -2 /mnt/temp_raid_1/tmp_1_2/ -r 16 -n 1 -u 64

The chia_plot command is using 16 cores, but the taskset tells it to only allow use of those 16 cores (the plotter itself will try to use more if it can). Worth noting that the OS sees 32 CPUs (0-31), but half of those are hyperthreading (16-31) and thus will suck capacity from the physical CPUs (0-15).

The "a" in -ac means "use this for this process, and all of its children, and the "c" means "use CPU numbers, not a binary mask".

**Reply by polkadotbabe** [reply]
Correct me if I'm wrong, but to use one thread of each 16 cores you'd want to use either all odd or even threads: taskset 0-31:2 

I'd think with modern hyperthreading it's not needed, and running on 31 of 32 threads would be nearly twice as fast as half the threads/one per core?

I have a bunch of 10K 300GB SAS drives sitting around, thinking of plotting off a RAID0 temps like you are, using four of them via PERC (Dell) controllers. If you have the time and desire, would you post the results of this fio run? Hardware or mdadm?

fio --filename=/home/sasraid0/1g_test --size=1GB --direct=1 --rw=randrw --bs=64k \\
--ioengine=libaio --iodepth=256   --runtime=120 --numjobs=4 \\
--time_based --group_reporting --name=iops-test-job --eta-newline=1

**Reply by digitalscream** [reply]
No, the siblings aren't sequential. Use this to figure it out:

 for file in /sys/devices/system/cpu/cpu[0-9]*/topology/thread_siblings_list; do echo -n "$file "; cat $file; done |sort -k2 -n

Which gives output like this (you can ignore duplicate lines):

/sys/devices/system/cpu/cpu0/topology/thread_siblings_list 0,4
/sys/devices/system/cpu/cpu1/topology/thread_siblings_list 1,5
/sys/devices/system/cpu/cpu2/topology/thread_siblings_list 2,6
/sys/devices/system/cpu/cpu7/topology/thread_siblings_list 3,7

So CPU0 is the physical core, and its corresponding HT is CPU4.

My system's actually set up for speedruns at the moment - just one big RAID0 array, plotting straight to the RAID5 destination array. And, annoyingly, I had a drive slightly fall out of my RAID5 DIY rack, so it's currently rebuilding the array :D

That said, my best so far was 1h19m for a single plot, which ain't bad for a system with no solid state storage at all :D

Oh, and I'm using hardware RAID with LSI controllers that have 512MB cache on them. I've got one with 1GB cache, but the damn thing's internal-only and I don't have enough space inside the machines for arrays this size.

**Reply by polkadotbabe** [reply]
How cool is that?! I never knew.  So for my ryzen 5 3600 desktop, six cores, 12 threads:

/sys/devices/system/cpu/cpu0/topology/thread_siblings_list 0,6
/sys/devices/system/cpu/cpu1/topology/thread_siblings_list 1,7
/sys/devices/system/cpu/cpu2/topology/thread_siblings_list 2,8
/sys/devices/system/cpu/cpu3/topology/thread_siblings_list 3,9
/sys/devices/system/cpu/cpu10/topology/thread_siblings_list 4,10
/sys/devices/system/cpu/cpu11/topology/thread_siblings_list 5,11

So the actual cores are 6-11, meaning the threads are really equal and are sequential: 0,0 1,1 2,2 and so on, meaning 0-5 are threads, 6-11 are actual cores (not threads)?

That is fast on spinny disks, yours are... 10K SAS? 

I plotted K34s on a RAID5 of SATA 5400rpm just for yucks, 48 hours each. Down to one hour with max's and a PCIe nvme as temp1+2 and the raid5 as destination... :-)

**Reply by digitalscream** [reply]
No, my interpretation is that 0-5 on yours are the physical CPUs, 6-11 are HT. To my knowledge, though, it doesn't really matter. As long as you're only using one of a pair, all of the physical core will be available.

Yes, my RAID0 is 8 x 10k SAS drives. I'm tempted to make the next one a 15k array, but I'm starting to hit the limits of what I'm willing to spend on it. I've had some drive failures, so I'm just waiting for another bunch to show up and then I'll see what it can do with 12 x 10k SAS RAID0.

The tmp1/destination array is actually 8 x SAS 7.2k 3TB drives in RAID5. It's surprisingly performant; in isolated tests, I can easily get 600MB/s+ out of it.

I've got both arrays on separate controllers, which seems a bit faster (makes sense, since each array gets its own 512MB write cache).

I suspect the reason I can get down to 1h20m isn't just the array speed, though; I can throw 16 physical cores at it. Oddly, the SAS drives actually end up faster using the 256GB 970 EVO NVMe drive I've got, and waaaaay faster than anything I can get with SATA SSDs.

**Reply by mostafa-aw** [reply]
> @mostafa-aw - this isn't Windows, it's Linux.
> 
> This is what I'm using, on a 2 x 8 core Xeon machine with SAS RAID0 as tmp2 and my SAS RAID5 destination array as tmp1:
> 
> taskset -ac 0-15 ~/bin/chia-plotter/build/chia_plot -t /mnt/plots_8/tmp_1/ -2 /mnt/temp_raid_1/tmp_1_2/ -r 16 -n 1 -u 64
> 
> The chia_plot command is using 16 cores, but the taskset tells it to only allow use of those 16 cores (the plotter itself will try to use more if it can). Worth noting that the OS sees 32 CPUs (0-31), but half of those are hyperthreading (16-31) and thus will suck capacity from the physical CPUs (0-15).
> 
> The "a" in -ac means "use this for this process, and all of its children, and the "c" means "use CPU numbers, not a binary mask".

Thanks for the explanation I'm using windows on a dual xeon e5 with 48 cores. Not getting good results.
That command u shared is for the madmax plotter, right?

**Reply by digitalscream** [reply]
Yes, it's for this plotter. Wouldn't be much point talking about the official plotter in these issues ;)

Your disks make a huge difference. For what it's worth, I'm plotting consistently at 3800-3850s in a single process using an old dual Xeon E5 2450 machine with a 14-drive 10k SAS RAID0 array as tmp2, and my RAID5 destination array as tmp1. Gonna be trying a 10k RAID0 array for tmp1 later this week, I reckon I can get below the hour with it, without wearing out any SSDs...

**Comment by 69chargr** [comment]
I'm going to try the overlap manually today and see what I can manage. The move to a permanent drive is going to slow this process down I believe to the point we are still at break even or worse compared to pechy combined + Swar. Just my guess though.

**Reply by ndaskalovic** [reply]
How did it go?

**Comment by digitalscream** [comment]
> I'm going to try the overlap manually today and see what I can manage. The move to a permanent drive is going to slow this process down I believe to the point we are still at break even or worse compared to pechy combined + Swar. Just my guess though.

For what it's worth, I wrote a quick bash script which loops the above config, but also spins the destination move into a separate process so the next plot can start immediately. It's not the most efficient way to do it, and it does slow down the start of the next plot, but overall it's faster.

If anyone needs to eke out a little more performance from their CPUs, then you should use cpufreq to force the CPU to turbo more (this works wonders on the E5 Xeons). I also disabled all kernel mitigations for CPU vulnerabilities, although that one comes with a massive disclaimer. My plotter has no Internet connectivity, and only SSH/VNC ports open for that reason.

**Reply by polkadotbabe** [reply]
I'd love to see the script, if you don't mind sharing?

Turbo, ugh, the heat. It's brutal, seeing cputemps spike up to 88c, so I turned it off... ...I get it, they're burst temps for a short duration, but are you worried?

So you're booting with mitigations=off in grub? If memory serves you don't have to recompile after kernel 5.2...

**Reply by digitalscream** [reply]
To be honest, the script is pretty much redundant after the latest updates. The only thing it gives you now is the ability to set CPU affinity for optimisation. I can send it to you for your curiosity, though.

As for temps, my Xeons are hitting 60C and no more. I might get an AIO in there, though, because the ambient heat in the case is starting to affect the SAS cards.

And yes, mitigations=off in grub with the latest kernel.

**Reply by polkadotbabe** [reply]
Unless something has changed, spectre/meltdown require existing account access, so it's not that big of a deal to turn off? I've never heard of a remote exploit for them? My machines are net-connected fulltime...

Odd, looking at temps with four staggered madmax plots -- Vs. the 12 concurrent I'd run before of Chia's plotter -- temps are way down, and overall plot production is up. Interesting... and not very intuitive.

I hadn't thought to use taskset, thanks for the tip above :-)

**Reply by mostafa-aw** [reply]
> > I'm going to try the overlap manually today and see what I can manage. The move to a permanent drive is going to slow this process down I believe to the point we are still at break even or worse compared to pechy combined + Swar. Just my guess though.
> 
> For what it's worth, I wrote a quick bash script which loops the above config, but also spins the destination move into a separate process so the next plot can start immediately. It's not the most efficient way to do it, and it does slow down the start of the next plot, but overall it's faster.
> 
> If anyone needs to eke out a little more performance from their CPUs, then you should use cpufreq to force the CPU to turbo more (this works wonders on the E5 Xeons). I also disabled all kernel mitigations for CPU vulnerabilities, although that one comes with a massive disclaimer. My plotter has no Internet connectivity, and only SSH/VNC ports open for that reason.

can u help with these xeons. i have 3 machines plotting and looking to optimize them?

**Reply by jfikar** [reply]
> If anyone needs to eke out a little more performance from their CPUs, then you should use cpufreq to force the CPU to turbo more (this works wonders on the E5 Xeons). 

@digitalscream, how exactly do you force turbo using cpufreq? Is is the same as `echo 0 | sudo tee /sys/devices/system/cpu/intel_pstate/no_turbo` ?

**Comment by delucca** [comment]
Well, you could easily create 2 bash scripts:

1. Launches the plotter WIHTOUT settings the `-d` param to keep the final plot in the temp memory
2. Listens to the temp dir and move everything ending with `plot` (without any other extension at the end) to the final dir

That would not slow down the start of the next plot

**Comment by madMAx43v3r** [comment]
The final destination copy runs parallel to the next plot, if you do more than 1 via `-n`.

**Comment by delucca** [comment]
> The final destination copy runs parallel to the next plot, if you do more than 1 via `-n`.

Amazing,

Man, you're a genius hahaha

So, no need to two bash scripts!

**Comment by digitalscream** [comment]
> That would not slow down the start of the next plot

It would still slow it down to a certain degree, if you're not using a ramdisk, because disk bandwidth is limited. Remember, this plotter isn't _only_ useful to people with insane amounts of available memory ;)

> The final destination copy runs parallel to the next plot, if you do more than 1 via `-n`.

Fair play, sir - I'm still using a slightly older version.

**Comment by madMAx43v3r** [comment]
Lol the plotter already does everything in parallel, would be stupid not to do the copy in parallel also ;)

**Comment by winkee01** [comment]
does it make things faster if I apply a raid0 with 2 ssd?

**Reply by polkadotbabe** [reply]
On my machines, no. Mind your fstrim frequency -- I run it once a day via cron and make sure you do it on the first format (which is counter-intuitive and took me a while to figure out why newly formatted NVMEs were slow) -- and on most of my machines xfs is faster, except for some reason on older machines where:

/dev/nvme1n1p1          /home/chia  ext4    noatime,nodiratime,commit=60,barrier=0,errors=remount-ro    0       1

Be mindful of those settings, not at all failure-tolerant  -- have your plotters on UPS/battery backup. 

 madmax's plotter seems much more cpu-dependent than NVME speed, and on all my machines, I end up with more plots just running one thread of chia_plot vs a bunch of parallel ones of chia's original plotter.

**Comment by digitalscream** [comment]
> does it make things faster if I apply a raid0 with 2 ssd?

Depends entirely on your system. The best way to find out is to try it, since the plotter is so fast that even if it doesn't work out, it hasn't really cost you much in the way of time :)

**Comment by madMAx43v3r** [comment]
if you don't use RAM and CPU is fast, probably yes

**Comment by MarcusArthur** [comment]
> does it make things faster if I apply a raid0 with 2 ssd?

I have 4x 1TB NVMEs in a RAID 0 config (totally overkill I know) and I'm able to get about 3000 seconds/finished plot.

**Comment by winkee01** [comment]
> > does it make things faster if I apply a raid0 with 2 ssd?
> 
> I have 4x 1TB NVMEs in a RAID 0 config (totally overkill I know) and I'm able to get about 3000 seconds/finished plot.

As a comparison, how much time will it take if no raid 0 is applied?

**Reply by Pit-70** [reply]
I'm running three 1 TB NVME Gen4 in a RAID 0 configuration. The performance is above 9.000 MB/s (read and write). A single NVME drive gives me about 3500 MB/s.

BUT: as this RAID is software based it also eats up some CPU power, which might lead to slower plotting performance.

As I'm still on Win10 I'm not able to run Madmax phantastic software, but looking forward for the Windows version

**Comment by mist0706** [comment]
I've done testing on the nvme raid0. Beyond 3000mb/s raid doesn't give any benefits. A single 3000mb/s nvme is as quick as two of the same in raid

**Reply by polkadotbabe** [reply]
I saw the same thing on PCIe 3.0 x4 with mdadm, PCIe 4 (with 4.0 nvmes of course) is wicked fast so never tried anything.

FIO benchmark showed the biggest speed-up was XFS (nearly 2x as fast as ext4 with largefile4 formatting) and scsi_mod.use_blk_mq=1 in grub. I think that's on by default in Ubuntu, it's not in CentOS or Gentoo. Although my Cent machines were 7-8-Stream live updates, which is supposed to be a no-no, so they might be funky compared to a fresh Cent install.

Cent uses XFS by default. Ubuntu is ext4.

**Comment by delucca** [comment]
I have a smaller rig with 2 NVMe of 512gb each

It seems that now my bottleneck is my CPU, since plotting both at the same time is slower than a single one

So, I'm probably going to raid those and try plotting larger k's

**Comment by maxpla3** [comment]
I have two question:
1. If I don’t use the ramdisk for tmpdir2 will the total TBW per plot be the same with this plotter as with the official one?
2. I have 128GB of RAM and a threadripper with 32 cores. With this setup, will it be possible to use 64 threads for one plotter AND use a ramdisk for tempdir2 at the same time? Will the 128GB of RAM be enough?

**Comment by NinjaDummie** [comment]
![Screenshot_84](https://user-images.githubusercontent.com/85762163/121733259-a32ffb00-cac9-11eb-9f04-1b6f582105fa.jpg)


There's something weird about SATA access.
Using a SATA SSD the plotter creates long CPU downtimes along with 100% SSD usage but very low transfer rate (5MB/s)

Doing the same test but using the SSD connected to the USB the result is normal. This way the Plotter can occupy high CPU rates and high SSD transfer rates (250MB/s)

I think there might be some problem with sata access which also occurs on the official plotter and which is alleviated with the USB cache.

**Comment by tobros91** [comment]
With threadripper 3970x I run 2 parallel, 30 threads per plot. 4 threads remaining is so that I still use to computer for programming. Locking affinity with taskset 0x000000007FFF7FFF for CPU 0-29 and 0x7FFF7FFF00000000 for CPU 32-62. Without locking affinity it was slower.

temp 1, x2 samsung 980 PRO raid 0
temp 2, 220GB ramdisk

Around 1700 seconds per plot, so a plot every 14 minutes.

![Screenshot from 2021-06-11 21-21-26](https://user-images.githubusercontent.com/4538330/121738847-1140e600-cafb-11eb-84be-821620feb989.png)

**Reply by polkadotbabe** [reply]
That's insanely fast!

**Reply by maxpla3** [reply]
That's around 100 plots per day, quite nice! If you reserve 220GB for ramdisk, 36GB remain for the 2x30 threads. Is that enough? According to madmax and other users here you need roughly 1GB of RAM per thread. I'm a bit confused.

**Reply by tobros91** [reply]
Ramdisk does not reserve the RAM, only uses what it needs. So staggering it never uses 220Gb at once. 

Not exactly sure what usual max usage is, but since plot completes with a +-20 seconds, 220GB leaves som room for errors.

**Reply by therealflinchy** [reply]
> That's around 100 plots per day, quite nice! If you reserve 220GB for ramdisk, 36GB remain for the 2x30 threads. Is that enough? According to madmax and other users here you need roughly 1GB of RAM per thread. I'm a bit confused.

it doesn't use 1gb per thread. depends on bucket size too. 512 buckets for me doesn't go past 10gb process ram usage for 32c for me.

**Reply by maxpla3** [reply]
How does the bucket size influence the performance? E.g. when you go from default to 512?

**Comment by mostafa-aw** [comment]
In the madmax plotter?

**Comment by mostafa-aw** [comment]
I'm new to these powershell commands. Do u mind sharing ur script?

**Reply by polkadotbabe** [reply]
I'm on unix, powershell is a windows scripting/shell, so don't know if this will help.

On unix, start by pasting: echo "$(nproc)"  in the shell and hitting enter. For me, that returned:

24

So there's 24 processors.

To keep the machine responsive, don't use all the cores, so "-r 22", and this will start it from the CLI:

/home/chia/chia_plot -r 22 -n -1 -b 256 -d /home/chia/ChiaP/ -p YOURPOOLKEY \\
-f YOURFARMERKEY -t /home/chia/temp1/ -2 /home/chia/ChiaTemp1/ 4 

Or, using the unix "screen" program to start another, detached terminal, and adding a "nice" value of +1 to further keep the machine responsive:

screen -d -m -S chia1 nice -n +1 /home/chia/chia_plot -r 22 -n -1 -b 256 \\
-d /home/chia/ChiaP/ -p YOURPOOLKEY \\
-f YOURFARMERKEY  -t /home/chia/temp1/ -2 /home/chia/ChiaTemp1/ 4 

chia keys show 
will tell you the appropriate keys.

screen -Dr chia1
will get you into that terminal. ctrl-a-d gets you out.

Hope this helps!

**Comment by therealflinchy** [comment]
so i've been attempting a p3 early start, concluded it's not worth worrying about.. it's either not possible because of insufficient ram, or should just do 2 plots parallel from start with your insanely high core count system w/256gb, or build a second system.

my system has 192gb of ram, i haven't watched p3 ramdisk usage closely (doing that now) but one in p3 one in p1, completely fills up a 170G ramdisk, which also doesn't leave enough system memory with with 512 buckets on 32c so the processes are paging anyway. so then i'd need 256gb of ram, which means i could split my processor affinity in half and just run two in parallel completely

or be more sensible and build another machine altogether. pretty disappointing, phase times weren't looking too heavily impacted by the stagger.

edit: yeah damn p3 ram usage up to like 90gb, that sucks. what a waste.

**Comment by isipisi89** [comment]
If you guys have 2 SSDS just run them in RAID 0 and set the chunk size to 512kb. It will be insane fast!
This plotter needs IO and CPU cors :D

**Reply by cyperbg** [reply]
> If you guys have 2 SSDS just run them in RAID 1 and set the chunk size to 512kb. It will be insane fast!
> This plotter needs IO and CPU cors :D

What is the benefit over Raid 0?
With Raid 1 writes are slower.

**Reply by isipisi89** [reply]
i mean 0 sorry was my fault.
